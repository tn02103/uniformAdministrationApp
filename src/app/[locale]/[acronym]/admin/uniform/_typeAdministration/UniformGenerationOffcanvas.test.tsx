import { UniformgenerationOffcanvas } from "@/app/[locale]/[acronym]/admin/uniform/_typeAdministration/UniformGenerationOffcanvas";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const sizeListIds = [
    'e667d674-7df8-436b-a2b8-77b06e063d36',
    'a961545b-28a7-409e-9200-1d85ccd53522',
    '07de1d59-4fc6-447b-98a6-da916e5792ef',
]
const uniformTypeId = '2b255769-2154-479f-bfca-c6a49e523334';
const testGeneration = {
    id: "testId",
    name: "Test Generation",
    outdated: false,
    fk_sizelist: sizeListIds[0],
    sortOrder: 0,
    sizelist: {
        id: sizeListIds[0],
        name: "Test Size List",
    },
}

// ################## MOCKS ##################
jest.mock("@/dataFetcher/uniformAdmin", () => {
    const typeListMutate = jest.fn(async (a) => a);
    return {
        useUniformSizelists: jest.fn(() => ({
            sizelistList: [{ id: sizeListIds[0], name: "Test Size List" }, { id: sizeListIds[1], name: "Test Size List 2" }, { id: sizeListIds[2], name: "Test Size List 3" }],
        })),
        useUniformTypeList: jest.fn(() => ({
            mutate: typeListMutate,
        })),
    };
});
jest.mock("@/dal/uniform/generation/_index", () => {
    return {
        createUniformGeneration: jest.fn(async () => "uniform generation created"),
        deleteUniformGeneration: jest.fn(async () => "uniform generation deleted"),
        updateUniformGeneration: jest.fn(async () => "uniform generation updated"),
    };
});

// ################## TESTS ##################
describe('<UniformgenerationOffcanvas>', () => {
    const { updateUniformGeneration, createUniformGeneration, deleteUniformGeneration } = jest.requireMock('@/dal/uniform/generation/_index');
    const { mutate } = jest.requireMock('@/dataFetcher/uniformAdmin').useUniformTypeList();
    const { useModal } = jest.requireMock('@/components/modals/modalProvider');
    const { dangerConfirmationModal } = useModal();
    const { toast } = jest.requireMock('react-toastify');

    afterEach(() => jest.clearAllMocks());

    describe('with generation', () => {
        it('renders the component', async () => {
            render(
                <UniformgenerationOffcanvas
                    generation={testGeneration}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            expect(screen.getByRole('dialog')).toMatchSnapshot();
        });
        it('hides sizelist if usingSizes is false', async () => {
            render(
                <UniformgenerationOffcanvas
                    generation={testGeneration}
                    uniformTypeId={uniformTypeId}
                    usingSizes={false}
                    onHide={() => { }}
                />
            );

            const sizeSelect = screen.queryByText('common.uniform.sizelist.label');
            expect(sizeSelect).not.toBeInTheDocument();
        });
        it('marks sizelist-label if usingSizes and sizelist is null', async () => {
            render(
                <UniformgenerationOffcanvas
                    generation={{ ...testGeneration, fk_sizelist: null }}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            const sizeSelect = screen.queryByText('common.uniform.sizelist.label');
            expect(sizeSelect).toBeInTheDocument();
            expect(sizeSelect).toHaveClass('text-danger');
        });
        it('sets editable state', async () => {
            const user = userEvent.setup();
            render(
                <UniformgenerationOffcanvas
                    generation={testGeneration}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            // get elements
            const editButton = screen.getByText('common.actions.edit');
            const deleteButton = screen.getByText('common.actions.delete');
            const nameInput = screen.getByRole('textbox', { name: 'common.name' });
            const sizeSelectDiv = screen.getByRole('paragraph', { name: 'common.uniform.sizelist.label' });
            const outdatedSwitch = screen.getByRole('switch', { name: 'common.uniform.generation.outdated' });
            // get variable elements
            let saveButton = screen.queryByText('common.actions.save');
            let cancelButton = screen.queryByText('common.actions.cancel');
            let sizeSelect = screen.queryByRole('combobox', { name: 'common.uniform.sizelist.label' });

            // check form components
            expect(nameInput).toBeVisible();
            expect(nameInput).toHaveClass('form-control-plaintext');
            expect(nameInput).toHaveAttribute('disabled');
            expect(outdatedSwitch).toBeVisible();
            expect(outdatedSwitch).toHaveAttribute('aria-disabled', 'true');
            expect(sizeSelect).toBeNull();
            expect(sizeSelectDiv).toBeVisible();

            // check buttons
            expect(editButton).toBeVisible();
            expect(deleteButton).toBeVisible();
            expect(saveButton).toBeNull();
            expect(cancelButton).toBeNull();
            expect(editButton).not.toHaveAttribute('disabled');
            expect(deleteButton).not.toHaveAttribute('disabled');

            // click edit button
            await user.click(editButton);
            saveButton = screen.queryByText('common.actions.save');
            cancelButton = screen.queryByText('common.actions.cancel');
            sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });

            // check form components
            expect(nameInput).toHaveClass('form-control');
            expect(nameInput).not.toHaveAttribute('disabled');
            expect(outdatedSwitch).toHaveAttribute('aria-disabled', 'false');
            expect(sizeSelect).toBeVisible();
            expect(sizeSelect).not.toHaveAttribute('disabled');
            expect(sizeSelectDiv).not.toBeVisible();

            // check buttons
            expect(editButton).toBeVisible();
            expect(deleteButton).toBeVisible();
            expect(saveButton).toBeVisible();
            expect(cancelButton).toBeVisible();
            expect(editButton).toHaveAttribute('disabled');
            expect(deleteButton).toHaveAttribute('disabled');
        });

        it('resets on cancel', async () => {
            const user = userEvent.setup();
            render(
                <UniformgenerationOffcanvas
                    generation={testGeneration}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            // set editable
            const editButton = screen.getByText('common.actions.edit');
            await user.click(editButton);

            // get elements
            const nameInput = screen.getByRole('textbox', { name: 'common.name' });
            const outdatedSwitch = screen.getByRole('switch', { name: 'common.uniform.generation.outdated' });
            const sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });
            const cancelButton = screen.getByText('common.actions.cancel');

            // change values
            await user.clear(nameInput);
            await user.type(nameInput, 'New Name');
            await user.click(outdatedSwitch);
            await user.selectOptions(sizeSelect, sizeListIds[1]);

            // check values
            expect(nameInput).toHaveValue('New Name');
            expect(outdatedSwitch).toHaveAttribute('aria-checked', 'true');
            expect(sizeSelect).toHaveValue(sizeListIds[1]);

            // cancel and check for resetet values again
            await user.click(cancelButton);
            expect(nameInput).toHaveValue(testGeneration.name);
            expect(outdatedSwitch).toHaveAttribute('aria-checked', String(testGeneration.outdated));
            expect(sizeSelect).not.toBeInTheDocument();
            const sizeSelectDiv = screen.getByRole('paragraph', { name: 'common.uniform.sizelist.label' });
            expect(sizeSelectDiv).toHaveTextContent(testGeneration.sizelist.name);
        });
    });
    describe('without generation', () => {
        it('renders in editable state if no generation is passed', async () => {
            render(
                <UniformgenerationOffcanvas
                    generation={null}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            expect(screen.getByRole('dialog')).toMatchSnapshot();
        });

        it('hides on Cancel', async () => {
            const user = userEvent.setup();
            const onHide = jest.fn();
            render(
                <UniformgenerationOffcanvas
                    generation={null}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={onHide}
                />
            );

            const cancelButton = screen.getByText('common.actions.cancel');
            await user.click(cancelButton);
            expect(onHide).toHaveBeenCalledTimes(1);
            expect(onHide).toHaveBeenCalledWith();
        });
    });
    describe('dal methods', () => {
        describe('delete', () => {
            it('deletes generation', async () => {
                const { dangerConfirmationModal } = useModal();
                // change mocks
                deleteUniformGeneration.mockReturnValue('uniform generation deleted');
                const onHide = jest.fn();

                // render component
                const user = userEvent.setup();
                render(
                    <UniformgenerationOffcanvas
                        generation={testGeneration}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // open danger confirmation modal
                const deleteButton = screen.getByText('common.actions.delete');
                await user.click(deleteButton);

                // validate parameters of dangerConfirmationModal
                expect(dangerConfirmationModal).toHaveBeenCalledTimes(1);
                expect(dangerConfirmationModal).toHaveBeenCalledWith({
                    header: 'admin.uniform.generationList.deleteModal.header',
                    message: expect.anything(),
                    confirmationText: 'admin.uniform.generationList.deleteModal.confirmationText',
                    dangerOption: {
                        option: 'common.actions.delete',
                        function: expect.any(Function),
                    },
                });

                // call delete function and validate actions
                await dangerConfirmationModal.mock.calls[0][0].dangerOption.function();
                expect(deleteUniformGeneration).toHaveBeenCalledTimes(1);
                expect(deleteUniformGeneration).toHaveBeenCalledWith(testGeneration.id);
                expect(mutate).toHaveBeenCalledTimes(1);
                expect(mutate).toHaveBeenCalledWith("uniform generation deleted");
                expect(onHide).toHaveBeenCalledTimes(1);
            });
            it('catches DAL-Exceptions', async () => {

                // set mocks
                deleteUniformGeneration.mockImplementationOnce(async () => { throw new Error("custom.error") });
                const onHide = jest.fn();

                // render component
                const user = userEvent.setup();
                render(
                    <UniformgenerationOffcanvas
                        generation={testGeneration}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // open danger confirmation modal
                const deleteButton = screen.getByText('common.actions.delete');
                await user.click(deleteButton);

                // validate parameters of dangerConfirmationModal
                expect(dangerConfirmationModal).toHaveBeenCalledTimes(1);

                // call delete function and validate actions                
                await dangerConfirmationModal.mock.calls[0][0].dangerOption.function();
                expect(deleteUniformGeneration).toHaveBeenCalledTimes(1);
                expect(mutate).toHaveBeenCalledTimes(1);
                expect(onHide).not.toHaveBeenCalled();

                // validate error toast
                expect(toast.error).toHaveBeenCalledTimes(1);
                expect(toast.error).toHaveBeenCalledWith("common.error.actions.delete");
            });
        });
        describe('create', () => {
            it('creates succefully', async () => {
                const user = userEvent.setup();
                const onHide = jest.fn();

                render(
                    <UniformgenerationOffcanvas
                        generation={null}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // get elements
                const nameInput = screen.getByRole('textbox', { name: 'common.name' });
                const outdatedSwitch = screen.getByRole('switch', { name: 'common.uniform.generation.outdated' });
                const sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });
                const createButton = screen.getByText('common.actions.create');

                // set values
                await user.type(nameInput, 'New Name');
                await user.click(outdatedSwitch);
                await user.selectOptions(sizeSelect, sizeListIds[1]);

                // check values
                expect(nameInput).toHaveValue('New Name');
                expect(outdatedSwitch).toHaveAttribute('aria-checked', 'true');
                expect(sizeSelect).toHaveValue(sizeListIds[1]);

                // create and validate actions
                await user.click(createButton);
                expect(createUniformGeneration).toHaveBeenCalledTimes(1);
                expect(createUniformGeneration).toHaveBeenCalledWith({
                    name: 'New Name',
                    outdated: true,
                    fk_sizelist: sizeListIds[1],
                    uniformTypeId: uniformTypeId,
                });

                expect(mutate).toHaveBeenCalledTimes(1);
                expect(mutate).toHaveBeenCalledWith("uniform generation created");
                expect(onHide).toHaveBeenCalledTimes(1);
            });
            it('catches form-errors', async () => {
                const user = userEvent.setup();
                createUniformGeneration.mockImplementationOnce(async () => {
                    return {
                        error: {
                            message: "custom.uniform.generation.nameDuplication",
                            formElement: "name",
                        }
                    };
                });

                render(
                    <UniformgenerationOffcanvas
                        generation={null}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={() => { }}
                    />
                );

                // get elements
                const nameInput = screen.getByRole('textbox', { name: 'common.name' });
                const sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });
                const createButton = screen.getByText('common.actions.create');

                // set values
                await user.type(nameInput, 'New Name');
                await user.selectOptions(sizeSelect, sizeListIds[0]);

                // creaet and validate error handling
                await user.click(createButton);
                expect(createUniformGeneration).toHaveBeenCalledTimes(1);
                expect(createUniformGeneration).toHaveBeenCalledWith({
                    name: 'New Name',
                    outdated: false,
                    fk_sizelist: sizeListIds[0],
                    uniformTypeId: uniformTypeId,
                });

                const errorMessage = await screen.findByText('custom.uniform.generation.nameDuplication');
                expect(errorMessage).toBeInTheDocument();
            });
            it('catches DAL-Exception', async () => {
                createUniformGeneration.mockImplementationOnce(async () => { throw new Error("custom.error") });
                const user = userEvent.setup();
                const onHide = jest.fn();

                render(
                    <UniformgenerationOffcanvas
                        generation={null}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // get elements
                const nameInput = screen.getByRole('textbox', { name: 'common.name' });
                const outdatedSwitch = screen.getByRole('switch', { name: 'common.uniform.generation.outdated' });
                const sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });
                const createButton = screen.getByText('common.actions.create');

                // set values and create
                await user.type(nameInput, 'New Name');
                await user.click(outdatedSwitch);
                await user.selectOptions(sizeSelect, sizeListIds[1]);
                await user.click(createButton);

                // validation exception handling
                expect(createUniformGeneration).toHaveBeenCalledTimes(1);
                expect(mutate).not.toHaveBeenCalledTimes(1);
                expect(onHide).not.toHaveBeenCalled();
                expect(toast.error).toHaveBeenCalledTimes(1);
                expect(toast.error).toHaveBeenCalledWith("common.error.actions.create");
            });
        });
        describe('update', () => {
            it('updates succefully', async () => {
                const user = userEvent.setup();
                const onHide = jest.fn();

                render(
                    <UniformgenerationOffcanvas
                        generation={testGeneration}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // set editable
                const editButton = screen.getByText('common.actions.edit');
                await user.click(editButton);

                // get elements
                const nameInput = screen.getByRole('textbox', { name: 'common.name' });
                const outdatedSwitch = screen.getByRole('switch', { name: 'common.uniform.generation.outdated' });
                const sizeSelect = screen.getByRole('combobox', { name: 'common.uniform.sizelist.label' });
                const saveButton = screen.getByText('common.actions.save');

                // change values
                await user.clear(nameInput);
                await user.type(nameInput, 'New Name');
                await user.click(outdatedSwitch);
                await user.selectOptions(sizeSelect, sizeListIds[1]);

                // check values
                expect(nameInput).toHaveValue('New Name');
                expect(outdatedSwitch).toHaveAttribute('aria-checked', 'true');
                expect(sizeSelect).toHaveValue(sizeListIds[1]);

                // save and validate actions
                await user.click(saveButton);
                expect(updateUniformGeneration).toHaveBeenCalledTimes(1);
                expect(updateUniformGeneration).toHaveBeenCalledWith({
                    data: {
                        name: 'New Name',
                        outdated: true,
                        fk_sizelist: sizeListIds[1],
                    },
                    id: testGeneration.id,
                });

                expect(mutate).toHaveBeenCalledTimes(1);
                expect(mutate).toHaveBeenCalledWith("uniform generation updated");
                expect(onHide).toHaveBeenCalledTimes(0);
            });
            it('catches form-errors', async () => {
                const user = userEvent.setup();
                updateUniformGeneration.mockImplementationOnce(async () => {
                    return {
                        error: {
                            message: "custom.uniform.generation.nameDuplication",
                            formElement: "name",
                        }
                    };
                });

                render(
                    <UniformgenerationOffcanvas
                        generation={testGeneration}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={() => { }}
                    />
                );

                // set editable
                const editButton = screen.getByText('common.actions.edit');
                await user.click(editButton);

                // get elements
                const nameInput = screen.getByRole('textbox', { name: 'common.name' });
                const saveButton = screen.getByText('common.actions.save');

                // setValues and save
                await user.clear(nameInput);
                await user.type(nameInput, 'New Name');
                await user.click(saveButton);

                // validate error handling
                expect(updateUniformGeneration).toHaveBeenCalledTimes(1);
                expect(updateUniformGeneration).toHaveBeenCalledWith({
                    data: {
                        name: 'New Name',
                        outdated: false,
                        fk_sizelist: sizeListIds[0],
                    },
                    id: testGeneration.id,
                });

                const errorMessage = await screen.findByText('custom.uniform.generation.nameDuplication');
                expect(errorMessage).toBeInTheDocument();
            });
            it('catches DAL-Exceptions', async () => {
                updateUniformGeneration.mockImplementationOnce(async () => { throw new Error("custom.error") });
                const user = userEvent.setup();
                const onHide = jest.fn();

                render(
                    <UniformgenerationOffcanvas
                        generation={testGeneration}
                        uniformTypeId={uniformTypeId}
                        usingSizes={true}
                        onHide={onHide}
                    />
                );

                // set editable and save
                await user.click(screen.getByText('common.actions.edit'));
                await user.click(screen.getByText('common.actions.save'));

                // validate exception handling
                expect(updateUniformGeneration).toHaveBeenCalledTimes(1);
                expect(mutate).not.toHaveBeenCalled();
                expect(onHide).not.toHaveBeenCalled();

                expect(toast.error).toHaveBeenCalledTimes(1);
                expect(toast.error).toHaveBeenCalledWith("common.error.actions.save");
            });
        });
    });
    describe('form validation', () => {
        it('shows error if name is empty', async () => {
            const user = userEvent.setup();
            render(
                <UniformgenerationOffcanvas
                    generation={testGeneration}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );
            // set editable
            const editButton = screen.getByText('common.actions.edit');
            await user.click(editButton);

            // clears name and clicks save
            const nameInput = screen.getByRole('textbox', { name: 'common.name' });
            const saveButton = screen.getByText('common.actions.save');
            await user.clear(nameInput);
            await user.click(saveButton);

            // validate error handling
            expect(updateUniformGeneration).toHaveBeenCalledTimes(0);
            const errorMessage = await screen.findByRole('alert', { name: 'error message name' });
            expect(errorMessage).toBeVisible();
            expect(errorMessage).toHaveTextContent('string.required');
        });

        it('should show error if sizelist is empty and usingSizes', async () => {
            const user = userEvent.setup();
            render(
                <UniformgenerationOffcanvas
                    generation={null}
                    uniformTypeId={uniformTypeId}
                    usingSizes={true}
                    onHide={() => { }}
                />
            );

            const nameInput = screen.getByRole('textbox', { name: 'common.name' });
            const createButton = screen.getByText('common.actions.create');
            await user.type(nameInput, 'New Name');
            await user.click(createButton);

            const sizelistError = screen.getByRole('alert', { name: 'error message fk_sizelist' });
            expect(sizelistError).toBeVisible();
            expect(sizelistError).toHaveTextContent('pleaseSelect');
        });
        it('should allow sizelist empty if not usingSizes', async () => {
            const user = userEvent.setup();
            render(
                <UniformgenerationOffcanvas
                    generation={null}
                    uniformTypeId={uniformTypeId}
                    usingSizes={false}
                    onHide={() => { }}
                />
            );

            const nameInput = screen.getByRole('textbox', { name: 'common.name' });
            const createButton = screen.getByText('common.actions.create');
            await user.type(nameInput, 'New Name');
            await user.click(createButton);

            expect(createUniformGeneration).toHaveBeenCalledTimes(1);
        });
    });
});
