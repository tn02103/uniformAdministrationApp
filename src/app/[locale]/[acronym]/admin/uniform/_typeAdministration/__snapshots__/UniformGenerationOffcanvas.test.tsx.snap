// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<UniformgenerationOffcanvas> with generation renders the component 1`] = `
<div
  aria-labelledby="generation-offcanvas-header"
  aria-modal="true"
  class="offcanvas offcanvas-end offcanvas-toggling show"
  role="dialog"
  style="width: 500px; visibility: visible;"
  tabindex="-1"
>
  <div
    class="offcanvas-header"
  >
    <div
      class="offcanvas-title h5"
    >
      <h2
        id="generation-offcanvas-header"
      >
        Test Generation
      </h2>
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      type="button"
    />
  </div>
  <div
    class="offcanvas-body"
  >
    <hr
      class="my-0"
    />
    <div
      class="mb-4 justify-content-evenly row"
    >
      <button
        class="col-auto border-0 btn btn-outline-secondary"
        type="button"
      >
        common.actions.edit
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pen fa-sm "
          data-icon="pen"
          data-prefix="fas"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M352.9 21.2L308 66.1 445.9 204 490.8 159.1C504.4 145.6 512 127.2 512 108s-7.6-37.6-21.2-51.1L455.1 21.2C441.6 7.6 423.2 0 404 0s-37.6 7.6-51.1 21.2zM274.1 100L58.9 315.1c-10.7 10.7-18.5 24.1-22.6 38.7L.9 481.6c-2.3 8.3 0 17.3 6.2 23.4s15.1 8.5 23.4 6.2l127.8-35.5c14.6-4.1 27.9-11.8 38.7-22.6L412 237.9 274.1 100z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="col-auto border-0 btn btn-outline-danger"
        type="button"
      >
        common.actions.delete
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-sm "
          data-icon="trash"
          data-prefix="fas"
          role="img"
          viewBox="0 0 448 512"
        >
          <path
            d="M136.7 5.9L128 32 32 32C14.3 32 0 46.3 0 64S14.3 96 32 96l384 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0-8.7-26.1C306.9-7.2 294.7-16 280.9-16L167.1-16c-13.8 0-26 8.8-30.4 21.9zM416 144L32 144 53.1 467.1C54.7 492.4 75.7 512 101 512L347 512c25.3 0 46.3-19.6 47.9-44.9L416 144z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <form>
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_input-name"
            >
              common.name
            </label>
            <input
              aria-invalid="false"
              class="form-control-plaintext"
              disabled=""
              id="uniformGenerationForm_input-name"
              name="name"
              type="text"
              value="Test Generation"
              width="auto"
            />
            <div
              aria-label="error message name"
              class="text-danger fs-7"
              data-testid="err_name"
              id="uniformGenerationForm_err_name"
              role="alert"
            />
          </div>
        </div>
        <div
          class="align-bottom col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_toggle-outdated"
            >
              common.uniform.generation.outdated
            </label>
            <div
              class="form-check form-switch px-0 my-auto"
            >
              <input
                aria-checked="false"
                aria-disabled="true"
                class="form-check-input ms-0 my-2"
                id="uniformGenerationForm_toggle-outdated"
                name="outdated"
                role="switch"
                style="pointer-events: none; opacity: 0.5;"
                tabindex="-1"
                type="checkbox"
                value="false"
              />
            </div>
            <div
              aria-label="error message outdated"
              class="text-danger fs-7"
              data-testid="err_outdated"
              id="uniformGenerationForm_err_outdated"
              role="alert"
            />
          </div>
        </div>
        <div
          class="col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_select-fk_sizelist"
            >
              common.uniform.sizelist.label
            </label>
            <p
              aria-label="common.uniform.sizelist.label"
              aria-readonly="true"
              class="py-2 m-0"
            >
              Test Size List
            </p>
            <div
              aria-label="error message fk_sizelist"
              class="text-danger fs-7"
              data-testid="err_fk_sizelist"
              id="uniformGenerationForm_err_fk_sizelist"
              role="alert"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<UniformgenerationOffcanvas> without generation renders in editable state if no generation is passed 1`] = `
<div
  aria-labelledby="generation-offcanvas-header"
  aria-modal="true"
  class="offcanvas offcanvas-end offcanvas-toggling show"
  role="dialog"
  style="width: 500px; visibility: visible;"
  tabindex="-1"
>
  <div
    class="offcanvas-header"
  >
    <div
      class="offcanvas-title h5"
    >
      <h2
        id="generation-offcanvas-header"
      >
        admin.uniform.generationList.header.create
      </h2>
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      type="button"
    />
  </div>
  <div
    class="offcanvas-body"
  >
    <hr
      class="my-0"
    />
    <div
      class="mb-4 justify-content-evenly row"
    >
      <button
        class="col-auto border-0 btn btn-outline-secondary"
        disabled=""
        type="button"
      >
        common.actions.edit
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pen fa-sm "
          data-icon="pen"
          data-prefix="fas"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M352.9 21.2L308 66.1 445.9 204 490.8 159.1C504.4 145.6 512 127.2 512 108s-7.6-37.6-21.2-51.1L455.1 21.2C441.6 7.6 423.2 0 404 0s-37.6 7.6-51.1 21.2zM274.1 100L58.9 315.1c-10.7 10.7-18.5 24.1-22.6 38.7L.9 481.6c-2.3 8.3 0 17.3 6.2 23.4s15.1 8.5 23.4 6.2l127.8-35.5c14.6-4.1 27.9-11.8 38.7-22.6L412 237.9 274.1 100z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="col-auto border-0 btn btn-outline-danger"
        disabled=""
        type="button"
      >
        common.actions.delete
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash fa-sm "
          data-icon="trash"
          data-prefix="fas"
          role="img"
          viewBox="0 0 448 512"
        >
          <path
            d="M136.7 5.9L128 32 32 32C14.3 32 0 46.3 0 64S14.3 96 32 96l384 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0-8.7-26.1C306.9-7.2 294.7-16 280.9-16L167.1-16c-13.8 0-26 8.8-30.4 21.9zM416 144L32 144 53.1 467.1C54.7 492.4 75.7 512 101 512L347 512c25.3 0 46.3-19.6 47.9-44.9L416 144z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <form>
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_input-name"
            >
              common.name
            </label>
            <input
              aria-invalid="false"
              class="form-control"
              id="uniformGenerationForm_input-name"
              name="name"
              type="text"
              value=""
              width="auto"
            />
            <div
              aria-label="error message name"
              class="text-danger fs-7"
              data-testid="err_name"
              id="uniformGenerationForm_err_name"
              role="alert"
            />
          </div>
        </div>
        <div
          class="align-bottom col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_toggle-outdated"
            >
              common.uniform.generation.outdated
            </label>
            <div
              class="form-check form-switch px-0 my-auto"
            >
              <input
                aria-checked="false"
                aria-disabled="false"
                class="form-check-input ms-0 my-2"
                id="uniformGenerationForm_toggle-outdated"
                name="outdated"
                role="switch"
                style="cursor: pointer;"
                tabindex="0"
                type="checkbox"
                value="false"
              />
            </div>
            <div
              aria-label="error message outdated"
              class="text-danger fs-7"
              data-testid="err_outdated"
              id="uniformGenerationForm_err_outdated"
              role="alert"
            />
          </div>
        </div>
        <div
          class="col-6"
        >
          <div>
            <label
              class="fw-bold m-0  form-label"
              for="uniformGenerationForm_select-fk_sizelist"
            >
              common.uniform.sizelist.label
            </label>
            <select
              aria-invalid="false"
              class="text-truncate  form-select"
              id="uniformGenerationForm_select-fk_sizelist"
              name="fk_sizelist"
            >
              <option
                disabled=""
                value=""
              >
                common.error.pleaseSelect
              </option>
              <option
                value="e667d674-7df8-436b-a2b8-77b06e063d36"
              >
                Test Size List
              </option>
              <option
                value="a961545b-28a7-409e-9200-1d85ccd53522"
              >
                Test Size List 2
              </option>
              <option
                value="07de1d59-4fc6-447b-98a6-da916e5792ef"
              >
                Test Size List 3
              </option>
            </select>
            <div
              aria-label="error message fk_sizelist"
              class="text-danger fs-7"
              data-testid="err_fk_sizelist"
              id="uniformGenerationForm_err_fk_sizelist"
              role="alert"
            />
          </div>
        </div>
        <div
          class="justify-content-evenly mt-2 mb-4 row"
        >
          <button
            class="col-auto btn btn-primary"
            type="submit"
          >
            common.actions.create
          </button>
          <button
            class="col-auto btn btn-outline-secondary"
            type="button"
          >
            common.actions.cancel
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
